### PR TITLE
RavenDB-21613 & RavenDB-21632 Create a range mapping before performing the actual aggregation to handle cases where there are no matches, resulting in empty ranges. | do not include null value in range query result

### PR DESCRIPTION
**Pull request by maciejaszyk**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-21613

### Type of change

- Bug fix

### How risky is the change?

- Not relevant

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing by Contributor

- Tests have been added that prove the fix is effective or that the feature works

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: test/SlowTests/Corax/RavenDB-21632.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+public class RavenDB_21632 : RavenTestBase
+{
+    public RavenDB_21632(ITestOutputHelper output) : base(output)
+    {
+    }
+    
+    [Fact]
+    public void CoraxFacetWillNotCountNullValuesInRangeFacets()
+    {
```
Comment (redknightlois) [inline test/SlowTests/Corax/RavenDB-21632.cs:21]: This should be run in both Lucene and Corax to ensure consistency.

Comment (arekpalinski) [reply test/SlowTests/Corax/RavenDB-21632.cs:21]: And please use `RavenFact`

File: test/SlowTests/Corax/RavenDB-21613.cs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                .AggregateBy(facet => facet.ByRanges(d => d.Age > 99))
+                .Execute();
+             
+            Assert.Equal(luceneResults.Values.First().Values.Count, coraxResults.Values.First().Values.Count);
+        }
+    }
+    
+    [RavenFact(RavenTestCategory.Querying | RavenTestCategory.Corax | RavenTestCategory.Facets)]
+    public void Test()
+    {
```
Comment (redknightlois) [inline test/SlowTests/Corax/RavenDB-21613.cs:49]: Use a better name that we can use to search.

File: src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexFacetedReadOperation.cs [hunk 3]
```diff
@@ -123,6 +125,40 @@ void UpdateRangeResults()
...
+
+                foreach (var range in result.Value.Ranges)
+                {
+                    var key = range.RangeText;
+                    if (facetValues.TryGetValue(key, out var collectionOfFacetValues))
+                        continue;
+
+                    collectionOfFacetValues = new FacetValues(facetQuery.Legacy);
+                    if ((result.Value.Aggregations.Count > 0) == false)
+                        collectionOfFacetValues.AddDefault(key);
```
Comment (redknightlois) [inline src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexFacetedReadOperation.cs:152]: Just stylistic, if you have braces on the else... would make sense to add braces on the if too.

**Review by redknightlois — APPROVED** [review]

**Comment by maciejaszyk** [comment]
Done.

File: test/SlowTests/Corax/RavenDB-21613.cs [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            
+            Dictionary<string, FacetResult> results = session
+                .Query<TestDocument>("indexes/test_document-lucene")
+                .AggregateBy(secondFacet)
+                .Execute();
+            
+            Dictionary<string, FacetResult> resultsCorax = session
+                .Query<TestDocument>("indexes/test_document-corax")
+                .AggregateBy(secondFacet)
+                .Execute();
```
Comment (arekpalinski) [inline test/SlowTests/Corax/RavenDB-21613.cs:80]: You don't need to do any assertion?

**Pull request merged**